pkg/test-infra/cdc/recommend.go

```diff
@@ -163,6 +170,147 @@ func newCDC(ns, name string) *CDC {
+		}
+		downstreamDB = fmt.Sprintf("%s-downstream-tidb", name)
+	)
+	logVol := corev1.VolumeMount{
+		Name:      "log",
+		MountPath: "/var/log/kafka",
+	}
+
+	return &Kafka{
+		Service: &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kafkaName,
+				Namespace: ns,
+				Labels:    kafkaLabels,
+			},
+			Spec: corev1.ServiceSpec{
+				Type: corev1.ServiceTypeClusterIP,
+				Ports: []corev1.ServicePort{
+					{
+						Name: "http",
+						Port: 9092,
+					},
+				},
+				ClusterIP: corev1.ClusterIPNone,
+				Selector:  kafkaLabels,
+			},
+		},
+		StatefulSet: &appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kafkaName,
+				Namespace: ns,
+				Labels:    kafkaLabels,
+			},
+			Spec: appsv1.StatefulSetSpec{
+				ServiceName: kafkaName,
+				Replicas:    pointer.Int32Ptr(1),
+				Selector:    &metav1.LabelSelector{MatchLabels: kafkaLabels},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{Labels: kafkaLabels},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "kafka",
+								Env: []corev1.EnvVar{
+									{
+										Name:  "KAFKA_MESSAGE_MAX_BYTES",
+										Value: "1073741824",
+									},
+									{
+										Name:  "KAFKA_REPLICA_FETCH_MAX_BYTES",
+										Value: "1073741824",
+									},
+									{
+										Name:  "KAFKA_ADVERTISED_PORT",
+										Value: "9092",
+									},
+									{
+										Name:  "KAFKA_ADVERTISED_HOST_NAME",
+										Value: "127.0.0.1",
+									},
+									{
+										Name:  "KAFKA_BROKER_ID",
+										Value: "1",
+									},
+									{
+										Name: "ZK", Value: "zk",
+									},
+									{
+										Name:  "KAFKA_ZOOKEEPER_CONNECT",
+										Value: "localhost:2181",
+									},
+								},
+								Image:           "wurstmeister/kafka",
+								ImagePullPolicy: corev1.PullIfNotPresent,
+								Ports: []corev1.ContainerPort{
+									{
+										Name:          "http",
+										ContainerPort: 9092,
+									},
+								},
+								VolumeMounts: []corev1.VolumeMount{logVol},
+							},
+						},
+						Volumes: []corev1.Volume{{
+							Name: "log",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						}},
+					},
+				},
+			},
+		},
+		Job: &batchv1.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kafkaJobName,
+				Namespace: ns,
+				Labels: map[string]string{
+					"app.kubernetes.io/name":      "tidb-cluster",
+					"app.kubernetes.io/component": "kafka",
+					"app.kubernetes.io/instance":  kafkaJobName,
+				},
+			},
+			Spec: batchv1.JobSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:            "cdc-kafka-consumer",
+								Image:           buildCDCImage("cdc-kafka-consumer"),
+								ImagePullPolicy: corev1.PullAlways,
+								Command: []string{
+									"/cdc_kafka_consumer",
```

Comment:
What's the cdc_kafka_consumer  function ?

Reply:
Pull data from kafka, decode and sink them to downstream tidb.